fix: undefined reference to CryptoPP::precomputedPrimeTable'

defunxyz opened the pull request:
**Summary:** #88.
According to the change log, Primetab.cpp has been added to the [cryptopp](https://github.com/weidai11/cryptopp) repository at the end of the June: [6ecc789](https://github.com/weidai11/cryptopp/commit/6ecc789df1cea7640f54ddc2aed149c6b188891f). But the last update of [0f506cf](https://github.com/abdes/cryptopp-cmake/commit/0f506cf5584422e8af255db7e333dec6727a9a9e), used in **cryptopp-cmake** project, is dated May 26, and this list does not include Primetab.cpp.

**Bug:** 
The **cryptopp-cmake** project was used to run _cmake_ on a separate project. This repository was added recently as a submodule to a separate project which contains **cryptopp** from [](https://github.com/weidai11/cryptopp) as a _submodule_ as well. Performing _cmake --build ._, and during LINKING stage of programs the issue of  `undefined reference to `CryptoPP::precomputedPrimeTable'` occurred:
```
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/13.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: lib/libcryptopp.a(nbtheory.cpp.obj):nbtheory.cpp:(.rdata$.refptr._ZN8CryptoPP21precomputedPrimeTableE[.refptr._ZN8CryptoPP21precomputedPrimeTableE]+0x0): undefined reference to `CryptoPP::precomputedPrimeTable'
collect2.exe: error: ld returned 1 exit status
mingw32-make[2]: *** [CMakeFiles\aes256.dir\build.make:101: bin/aes256.exe] Error 1
mingw32-make[1]: *** [CMakeFiles\Makefile2:215: CMakeFiles/aes256.dir/all] Error 2
mingw32-make: *** [Makefile:135: all] Error 2
```

**Fix:**
This issue is resolved by adding the `primetab.cpp` to line 120 in `cryptopp/sources.cmake` file. 

**Primetab.cpp:** 
```
// primetab.cpp - written and placed in the public domain by Jeffrey Walton
//
// nbtheory.cpp originally built the prime table on the fly, and then
// returned a reference to it in a Singleton. This was useful to
// save memory in the old days. Nowadays we can precompute the table
// and place it in the read-only data segment. Also see
// https://github.com/weidai11/cryptopp/issues/1210.

// The table below was generated from the original Crypto++ table:
```

abdes commented:
Thank you for suggesting this PR.
It is a known change and as per instructions of the crypto++ maintainer, this file is to be added only when the 8.9.0 is released.

defunxyz commented:
Hi, 
yes, I seen this information. However, I added their recent available push as a submodule to a separate project of mine (see [here](https://github.com/defunxyz/cryptoutils)). And I kept receiving that issue only once I added the  `primetab.cpp `to line **120** in `cryptopp/sources.cmake` file. That the issue was resolve for me.

Vollstrecker commented:
As we aim on stable and this one will happen in the next release, we COULD add it guarded by if(CRYPTOPP_USE_MASTER)

abdes commented:
> As we aim on stable and this one will happen in the next release, we COULD add it guarded by if(CRYPTOPP_USE_MASTER)

Except that we do not build with master, we only build with stable releases of crypto++. As you can see from the checks, all builds will fail cause we only pull the tracked release.

Given that historically it was somewhat a strong assumption that we just track released versions of crypto++, I'd like to get @noloader opinion first before changing this strong assumption.

Vollstrecker commented:
Someone seemed to have another oppinion: [see her](https://github.com/abdes/cryptopp-cmake/blame/3afc4c53be95c72ac7efb7aa9a92c548b489cd66/CMakeLists.txt#L131)

Interesting that blame says it's been in the first commit here, but in the old one I can't find it.

abdes commented:
I'm sorry, I wrote the reply hastily as I did not have a lot of time. Let me explain myself better...

You can pull the sources of `cryptopp` from the master branch using `CRYPTOPP_USE_MASTER_BRANCH` but that is intended to test fixes and release compatible changes that may have been made to the `cryptopp` sources and have not yet been integrated into a fix release of `cryptopp-cmake`. This of course only works if you are using git to get the cryptopp sources and not an archive file of the released sources for example.

It has never been intended to jump ahead of `cryptopp` version neither in the past, not in this modern fork of `cryptopp-cmake`.  The reason is quite simple: the maintenance load of keeping the two projects in sync. The compromise is a constraint on the versioning scheme of `cyptopp-cmake` and a having to sometimes wait until a release of `crtyopp` happens before you can get the latest features.

I hope this explains better the philosophy and the history.

Now, my opinion is quite clear: I am not in favor of having out of release files added or flags added/changed as I do not have confidence that over time we will have enough bandwidth and discipline to keep things in sync. As this is opensource, I am open of course to multiple opinions and that's why I wanna hear from @noloader,as well. And because this opensource, we also need to have committers that can dedicate time and bandwidth to maintain things after a PR.

If we decide to go this path, I'd like to get a more elaborate scheme for this PR for how to manage out of release changes like this one without losing track of the changes and their lifecycle. Something that could look like an addon to the standard build like what we do when we apply a patch before we build a package...

abdes commented:
Changes made in #88 

Thank you for the initial contribution.
